sktime/forecasting/hf_patchtst_forecaster.py

```diff
@@ -0,0 +1,548 @@
+        ],
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "scitype:y": "both",
+        "ignores-exogeneous-X": True,
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "capability:pred_int": False,
+        "capability:pred_int:insample": False,
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "python_dependencies": ["transformers", "torch"],
+        "capability:global_forecasting": True,
+    }
+
+    def __init__(
+        self,
+        # model variables except for forecast_columns
+        model_path=None,
+        fit_strategy="full",
+        validation_split=0.2,
+        config=None,
+        training_args=None,
+        compute_metrics=None,
+        callbacks=None,
+    ):
+        self.model_path = model_path
+        self.fit_strategy = fit_strategy
+        # dataset and training parameters
+        self.validation_split = validation_split
+        self.config = config
+        self.training_args = training_args
+        self.compute_metrics = compute_metrics
+        self.callbacks = callbacks
+
+        self._config = deepcopy(self.config)
+        super().__init__()
+        if self.fit_strategy not in ["full", "minimal", "zero-shot"]:
+            raise ValueError("unexpected fit_strategy passed in argument")
+
+    def _fit(self, y, fh, X=None):
+        """Fits the model.
+
+        Parameters
+        ----------
+        y : pandas DataFrame
+            pandas dataframe containing single or multivariate data
+
+        fh : Forecasting Horizon object
+            used to determine forecasting horizon for predictions
+
+        Returns
+        -------
+        self : a reference to the object
+        """
+        self._y = y
+        self.fh_ = int(max(fh.to_relative(self.cutoff)))
+        self.y_columns = y.columns
+        # if no model_path was given, initialize new full model from config
+        if not self.model_path:
+            self._config["num_input_channels"] = len(self.y_columns)
+            if "prediction_length" in self._config.keys():
+                if self._config["prediction_length"] > self.fh_:
+                    warnings.warn(
+                        "Found prediction length inside config larger"
+                        "than passed fh, will use larger of the two"
+                    )
+                elif self._config["prediction_length"] <= self.fh_:
+                    warnings.warn(
+                        "Found fh length inside config larger"
+                        "than passed prediction_length, will use larger of the two"
+                    )
+                    self._config["prediction_length"] = int(self.fh_)
+            else:
+                self._config["prediction_length"] = int(self.fh_)
```

Comment:
this has already been cast to integer right? then why do again?

Reply:
there was some strange issue where `self.fh_` was being converted into a numpy integer which is not json serializable, I've opted to just call `int()` to ensure it is being casted to an integer